features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/stored-procedures/db2/sessiondata-cleanup/db2-session-data-cleanup.sql

```diff
@@ -0,0 +1,195 @@
+	DECLARE @operationalSessionAppInfoCleanupCount INT;
+	DECLARE @sessionMetadataCleanupCount INT;
+	DECLARE @operationalSessionMetadataCleanupCount INT;
+	DECLARE @operationCleanupCount INT;
+	DECLARE @tracingEnabled INT;
+	DECLARE @sleepTime VARCHAR(8);
+	DECLARE @batchSize INT;
+	DECLARE @chunkLimit INT;
+
+	DECLARE @sessionCleanUpTempTableCount INT;
+	DECLARE @operationCleanUpTempTableCount INT;
+	DECLARE @cleanUpCompleted INT;
+	DECLARE @autocommit INT;
+	DECLARE @sessionCleanupTime bigint;
+	DECLARE @operationCleanupTime bigint;
+	DECLARE @OLD_SQL_SAFE_UPDATES INT;
+	DECLARE @SQL_SAFE_UPDATES INT;
+
+	DECLARE @message VARCHAR(32672);
+	DECLARE @status integer default 0;
+
+	-- ------------------------------------------
+    -- CONFIGURABLE VARIABLES
+    -- ------------------------------------------
+	SET @batchSize = 5000;
+	-- This defines the number of entries from IDN_AUTH_SESSION_STORE that are taken into a SNAPSHOT
+	SET @chunkLimit=1000000;
+	SET @deletedSessions = 0;
+	SET @deletedUserSessionMappings = 0;
+	SET @deletedOperationalUserSessionMappings = 0;
+	SET @deletedSessionAppInfo = 0;
+	SET @deletedOperationalSessionAppInfo = 0;
+	SET @deletedSessionMetadata = 0;
+	SET @deletedOperationalSessionMetadata = 0;
+	SET @deletedStoreOperations = 0;
+	SET @deletedDeleteOperations = 0;
+	SET @sessionCleanupCount = 1;
+	SET @sessionMappingsCleanupCount = 1;
+	SET @operationalSessionMappingsCleanupCount = 1;
+	SET @sessionAppInfoCleanupCount = 1;
+	SET @operationalSessionAppInfoCleanupCount = 1;
+	SET @sessionMetadataCleanupCount = 1;
+	SET @operationalSessionMetadataCleanupCount = 1;
+	SET @operationCleanupCount = 1;
+	SET @tracingEnabled = 1;	-- SET IF TRACE LOGGING IS ENABLED [DEFAULT : FALSE]
+	SET @sleepTime = '00:00:02.000';          -- Sleep time in seconds.
+	SET @autocommit = 0;
+
+	SET @sessionCleanUpTempTableCount = 1;
+	SET @operationCleanUpTempTableCount = 1;
+	SET @cleanUpCompleted = 1;
+
+	-- Session data older than 20160 minutes(14 days) will be removed.
+	SET @sessionCleanupTime = BIGINT(CURRENT TIMESTAMP - 14 DAYS);
+	-- Operational data older than 720 minutes(12 h) will be removed.
+	SET @operationCleanupTime = BIGINT(CURRENT TIMESTAMP - (12) HOUR);
+
+	SET @SQL_SAFE_UPDATES = 0;
+	SET @OLD_SQL_SAFE_UPDATES=@SQL_SAFE_UPDATES;
+
+	-- ------------------------------------------
+	-- REMOVE SESSION DATA
+	-- ------------------------------------------
+
+	-- CLEANUP ANY EXISTING TEMP TABLES
+	DROP TABLE IF EXISTS IDN_AUTH_SESSION_STORE_TMP;
+	DROP TABLE IF EXISTS TEMP_SESSION_BATCH;
+	COMMIT;
+
+	-- RUN UNTIL
+	WHILE (@sessionCleanUpTempTableCount > 0) DO
+		IF NOT (EXISTS (SELECT NAME FROM SYSIBM.SYSTABLES WHERE NAME= 'IDN_AUTH_SESSION_STORE_TMP')) THEN
+		    CREATE TABLE IDN_AUTH_SESSION_STORE_TMP AS (SELECT SESSION_ID FROM IDN_AUTH_SESSION_STORE) WITH NO DATA;
+			INSERT INTO IDN_AUTH_SESSION_STORE_TMP SELECT SESSION_ID FROM IDN_AUTH_SESSION_STORE WHERE TIME_CREATED < @sessionCleanupTime LIMIT @chunkLimit;
```

Comment:
Shall we change this TIME_CREATED value with EXPIRY_TIME value

Reply:
This will be completed as a separate task as the change needs to be done in other scripts (MSSQL, Oracle etc. ) as well. 
@Kanapriya